src/moonlink/src/storage/mooncake_table.rs

```diff
@@ -332,15 +333,83 @@ impl IcebergPersistedRecords {
     pub fn get_file_indices_to_reflect_persistence(&self) -> (HashSet<FileId>, Vec<FileIndex>) {
         let mut persisted_file_indices = vec![];
         persisted_file_indices.extend(self.import_result.new_file_indices.iter().cloned());
+        persisted_file_indices.extend(
+            self.index_merge_result
+                .new_file_indices_imported
+                .iter()
+                .cloned(),
+        );
+        persisted_file_indices.extend(
+            self.data_compaction_result
+                .new_file_indices_imported
+                .iter()
+                .cloned(),
+        );
+
+        let mut index_blocks_to_delete = HashSet::new();
```

Comment:
Similar to above, index blocks to delete should also consider all imported file indices.